### PR TITLE
Added dependency of perlbrewr and some minor edits to documentation

### PR DESCRIPTION
**Pull request by fontikar**
We needed perlbrewr so R can find the right copy of perl. 
Minor changes to documentation to get R CMD CHECK passing

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -2,4 +2,12 @@
 
 export(analyse_oneshot_spatial)
 export(analyse_rasters_spatial)
+export(basedata)
+export(install_perlbrewr)
+export(load_data_)
```
Comment (shawnlaffan) [inline NAMESPACE:7]: Not all of these need to be exported.  

Utility funcs like ```install_perlbrewr``` can be exported but any data analysis needs to go via the R6 basedata object.

File: NAMESPACE [hunk 2]
```diff
@@ -2,4 +2,12 @@
...
+export(process_tabular_results)
+export(run_cluster_analysis)
+export(run_randomisation_analysis)
+export(run_spatial_analysis)
 export(start_server)
+importFrom(perlbrewr,perlbrew)
```
Comment (shawnlaffan) [inline NAMESPACE:13]: Probably for later, but we only need perlbrewr on non-Windows machines.  We have Strawberry perl on Windows, and perlbrew does not work on Windows.

File: R/install_perl_on_mac.R [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#' Set Perl up on macOS
```
Comment (shawnlaffan) [inline R/install_perl_on_mac.R:1]: This can be used on any non-Windows, unix-derived system that we support (mac, linux, BSD variants etc).

**Review by shawnlaffan — CHANGES_REQUESTED** [review]

File: NAMESPACE (hunk 2 above)
Comment (fontikar) [reply NAMESPACE:13]: Unfortunately, R packages does not allow OS specific dependencies. I think this is ok for now, and perlbrewr is pretty lite

File: NAMESPACE (hunk 1 above)
Comment (fontikar) [reply NAMESPACE:7]: Which ones would you like to export? :) Do you want to change that from your end by removing the roxygentag @export

File: R/install_perl_on_mac.R (hunk 3 above)
Comment (fontikar) [reply R/install_perl_on_mac.R]: Just dated this one!

File: NAMESPACE (hunk 1 above)
Comment (fontikar) [reply NAMESPACE:7]: If you tell me which ones I can also change :)

Comment (shawnlaffan) [reply NAMESPACE:7]: ```basedata``` and ```install_perlbrewr``` might be all we need from the set being added in this PR.  

The ```run_*``` functions all require book-keeping that the ```basedata``` object handles.  ```process_tabular_results``` is there to convert data from JSON to tabular form.

**Comment by fontikar** [comment]
Will wait for your green light @shawnlaffan 🟢  Feel free to do the merge in your time!

**Pull request merged**